libsplinter/src/circuit/template/mod.rs

```diff
@@ -102,25 +112,30 @@ impl CircuitTemplateManager {
 
     /// Lists all available circuit templates found in the `path` of the `CircuitTemplateManager`.
     pub fn list_available_templates(&self) -> Result<Vec<String>, CircuitTemplateError> {
-        let path = Path::new(&self.path);
-        let available_templates = path
-            .read_dir()
-            .map_err(|err| {
-                CircuitTemplateError::new_with_source(
-                    &format!("Failed to read circuit template files in {}", self.path),
-                    Box::new(err),
-                )
-            })?
+        let template_pattern = String::from("*.yaml");
+        let available_templates = self
+            .paths
+            .iter()
+            .map(|path| {
+                let template_path = Path::new(path).join(&template_pattern);
+                let pattern_string = template_path.to_str().ok_or_else(|| {
+                    CircuitTemplateError::new(&String::from("Template path is not valid UTF-8"))
+                })?;
+                Ok(glob(pattern_string).map_err(|_| {
+                    CircuitTemplateError::new(&format!(
+                        "Cannot query path {:?} for pattern: {}",
+                        template_path, pattern_string
```

Comment:
You should be able to use template_path.display() here instead of converting it back to a string, avoiding the potential for the first CircuitTemplateError in this function.